File: Data Structures & Algorithms/two-integer-sum-ii/submission-2.java
class Solution {
    public int[] twoSum(int[] numbers, int target) {
        var output = new int [2];
        int i = 0, j = numbers.length - 1;
        while ( i < j ){
            if (numbers[i] + numbers[j] == target){
                output[0] = i + 1;
                output[1] = j + 1;
                return output;
            }
            else if (numbers[i] + numbers[j] < target){
                i++;
            }
            else{
                j--;
            }
        }

        return output;
    }
}

File: Data Structures & Algorithms/combination-target-sum-ii/submission-2.py
class Solution:
    def combinationSum2(self, candidates: List[int], target: int) -> List[List[int]]:
        res = []
        cur = []
        candidates.sort()
        def dfs(target, index):
            if target < 0:
                return 
            
            if target == 0:
                res.append(cur.copy())
                return
            
            for i in range(index, len(candidates)):
                #如果当前数字和前一个数字一样，并且它不是这一层的第一个选择，就跳过
                if i > index and candidates[i] == candidates[i - 1]:
                    continue
                
                cur.append(candidates[i])
                dfs(target - candidates[i], i + 1)
                cur.pop()
        
        dfs(target, 0)
        return res
